File: frontend/src/api/weather.ts
import type { WeatherInfo } from "../types/weather";
import { api } from "./client";

export function getWeather(city?: string): Promise<WeatherInfo> {
  const query = city ? `?city=${encodeURIComponent(city)}` : "";
  return api<WeatherInfo>(`/api/weather${query}`);
}

File: backend/app/api/weather.py
from fastapi import APIRouter, Depends

from ..auth.cognito import CurrentUser, get_current_user
from ..services import weather

router = APIRouter()


@router.get("/api/weather")
def get_weather(city: str | None = None, user: CurrentUser = Depends(get_current_user)):
    return weather.get_weather(city)
